stable/admin/README.md

```diff
@@ -170,8 +170,15 @@ The following tables list the configurable parameters for the `admin` option of
 | `logPersistence.size` | Amount of disk space allocated for log storage | `60Gi` |
 | `logPersistence.storageClass` | Storage class for volume backing log storage | `-` |
 | `envFrom` | Import ENV vars from one or more configMaps | `[]` |
-| `options` | Set optons to be passed to nuoadmin as arguments | `{}` |
-| `securityContext.capabilities` | add capabilities to the container | `[]` |
+| `options` | Set options to be passed to nuoadmin as arguments | `{}` |
+| `initContainers.runInitDisk` | Whether to run the `init-disk` init container to set volume permissions | `true` |
+| `initContainers.runInitDiskAsRoot` | Whether to run the `init-disk` init container as root | `true` |
+| `securityContext.fsGroupOnly` | Creates a security context for Pods containing only the `securityContext.fsGroup` value | `false` |
+| `securityContext.runAsNonRootGroup` | Creates a security context for Pods containing a non-root user and group (1000:1000) along with the `securityContext.fsGroup` value | `false` |
+| `securityContext.enabled` | Creates a security context for Pods containing the `securityContext.runAsUser` and `securityContext.fsGroup` values | `false` |
```

Comment:
My understanding is that `securityContext.enabled`, `securityContext.fsGroupOnly` and `securityContext.runAsNonRootGroup` are mutual exclusive options. Would it be better if we have something like below?

```yaml
securityContext:
  # disable securityContext
  enabled: false
  # One of runAsRootGroup|runAsNonRootGroup|fsGroupOnly
  runAsPreset: "runAsRootGroup"
  runAsUser: 1000
  runAsGroup: 0
  fsGroup: 1000
  ...
```

Reply:
The reason why I went with this approach is that it I didn't see any pattern for validating options for cases where one of the well-known constants is misspelled. I guess the user could also misspell the key, e.g. `--set admin.securityContext.runAsNonRootGropu=true`, and that would also silently fail because the values file does not have a schema (that I'm aware of).

I still have a slight preference for this approach because it allows the user to enable the different modes by setting only one value, assuming that the values file does not set any of these values to `true`. For example, to enable `fsGroup`, the user can do

```
... --set admin.securityContext.fsGroupOnly=true
```

as opposed to

```
... --set admin.securityContext.enabled=true --set admin.securityContext.runAsPreset=fsGroupOnly
```

Reply:
Works for me.

> I didn't see any pattern for validating options for cases where one of the well-known constants is misspelled

I don't think that Helm can do schema validation. We have poor man's logic for validating `autoRestore.source`:

```
{{- define "autoRestore.source" -}}
{{- if hasPrefix ":" .Values.database.autoRestore.source }}
{{- $valid := list ":latest" ":group-latest" "" }}
{{- if not (has .Values.database.autoRestore.source $valid) }}
{{- fail (printf "Invalid autorestore source: %s" .Values.database.autoRestore.source) }}
{{- end }}
{{- end -}}
{{- with .Values.database.autoRestore.source }}
{{- . }}
{{- end -}}
{{- end -}}
```